<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GP Developer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/icons/application-icon.png</normaloff>:/icons/application-icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="titleWidget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap resource="icons.qrc">:/icons/application-icon.png</pixmap>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="title">
         <property name="styleSheet">
          <string notr="true">QLabel { font-size: 18px; }</string>
         </property>
         <property name="text">
          <string>GP Developer</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>294</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="Developer::QuickRunWidget" name="quickRunWidget" native="true"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="Developer::PrettyTabWidget" name="tabWidget" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuNew">
     <property name="toolTip">
      <string>Create a new project or file</string>
     </property>
     <property name="title">
      <string>New...</string>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/small_page_add.png</normaloff>:/icons/small_page_add.png</iconset>
     </property>
     <addaction name="actionNewProject"/>
     <addaction name="actionNewProgram"/>
     <addaction name="actionNewRule"/>
     <addaction name="actionNewGraph"/>
    </widget>
    <widget class="QMenu" name="menuOpen">
     <property name="title">
      <string>Open/Import...</string>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/small_open_folder.png</normaloff>:/icons/small_open_folder.png</iconset>
     </property>
     <addaction name="actionOpenProject"/>
     <addaction name="actionOpenGraph"/>
    </widget>
    <widget class="QMenu" name="menuRecentProjects">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Recent Projects</string>
     </property>
     <property name="icon">
      <iconset theme="document-open-recent">
       <normaloff/>
      </iconset>
     </property>
    </widget>
    <addaction name="menuNew"/>
    <addaction name="menuOpen"/>
    <addaction name="menuRecentProjects"/>
    <addaction name="separator"/>
    <addaction name="actionCloseProject"/>
    <addaction name="separator"/>
    <addaction name="actionSaveCurrentFile"/>
    <addaction name="actionSaveCurrentFileAs"/>
    <addaction name="actionSaveAll"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuFindReplace">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Find/Replace</string>
     </property>
     <property name="icon">
      <iconset resource="icons.qrc">
       <normaloff>:/icons/small_find.png</normaloff>:/icons/small_find.png</iconset>
     </property>
     <addaction name="actionReplaceInCurrentFile"/>
     <addaction name="actionReplaceInAllFiles"/>
    </widget>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionSelectAll"/>
    <addaction name="separator"/>
    <addaction name="menuFindReplace"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <widget class="QMenu" name="menuLayout">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Layout</string>
     </property>
     <addaction name="actionLayoutSugiyama"/>
     <addaction name="actionLayoutCircular"/>
    </widget>
    <addaction name="menuLayout"/>
    <addaction name="separator"/>
    <addaction name="actionOptions"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionApplicationHelp"/>
    <addaction name="separator"/>
    <addaction name="actionAboutApplication"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionApplicationHelp">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_bookmark.png</normaloff>:/icons/small_bookmark.png</iconset>
   </property>
   <property name="text">
    <string>Application Help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionAboutApplication">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/application-icon.png</normaloff>:/icons/application-icon.png</iconset>
   </property>
   <property name="text">
    <string>About GP Developer</string>
   </property>
  </action>
  <action name="actionNewProject">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_folder.png</normaloff>:/icons/small_folder.png</iconset>
   </property>
   <property name="text">
    <string>GP Project</string>
   </property>
   <property name="toolTip">
    <string>Create a new GP Project</string>
   </property>
   <property name="statusTip">
    <string>Create a new GP Project</string>
   </property>
  </action>
  <action name="actionNewProgram">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_page.png</normaloff>:/icons/small_page.png</iconset>
   </property>
   <property name="text">
    <string>Program</string>
   </property>
  </action>
  <action name="actionNewRule">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_brick.png</normaloff>:/icons/small_brick.png</iconset>
   </property>
   <property name="text">
    <string>Rule</string>
   </property>
  </action>
  <action name="actionNewGraph">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/application-icon.png</normaloff>:/icons/application-icon.png</iconset>
   </property>
   <property name="text">
    <string>Graph</string>
   </property>
  </action>
  <action name="actionOpenProject">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_folder.png</normaloff>:/icons/small_folder.png</iconset>
   </property>
   <property name="text">
    <string>GP Project</string>
   </property>
  </action>
  <action name="actionOpenGraph">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/application-icon.png</normaloff>:/icons/application-icon.png</iconset>
   </property>
   <property name="text">
    <string>Graph</string>
   </property>
  </action>
  <action name="actionCloseProject">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Close Project</string>
   </property>
  </action>
  <action name="actionSaveCurrentFile">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_page_save.png</normaloff>:/icons/small_page_save.png</iconset>
   </property>
   <property name="text">
    <string>Save File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveCurrentFileAs">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_save_as.png</normaloff>:/icons/small_save_as.png</iconset>
   </property>
   <property name="text">
    <string>Save File As...</string>
   </property>
  </action>
  <action name="actionSaveAll">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_page_save.png</normaloff>:/icons/small_page_save.png</iconset>
   </property>
   <property name="text">
    <string>Save All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_cancel.png</normaloff>:/icons/small_cancel.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_arrow_undo.png</normaloff>:/icons/small_arrow_undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_arrow_redo.png</normaloff>:/icons/small_arrow_redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_cut.png</normaloff>:/icons/small_cut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_page_copy.png</normaloff>:/icons/small_page_copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_paste.png</normaloff>:/icons/small_paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_page_white_text.png</normaloff>:/icons/small_page_white_text.png</iconset>
   </property>
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionReplaceInCurrentFile">
   <property name="text">
    <string>Current File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionReplaceInAllFiles">
   <property name="text">
    <string>All Files</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
  <action name="actionLayoutSugiyama">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Sugiyama</string>
   </property>
  </action>
  <action name="actionLayoutCircular">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Circular</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="icon">
    <iconset resource="icons.qrc">
     <normaloff>:/icons/small_wrench.png</normaloff>:/icons/small_wrench.png</iconset>
   </property>
   <property name="text">
    <string>Options</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Developer::PrettyTabWidget</class>
   <extends>QWidget</extends>
   <header>prettytabwidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Developer::QuickRunWidget</class>
   <extends>QWidget</extends>
   <header>quickrunwidget.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionApplicationHelp</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showApplicationHelp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewProject</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>newProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAboutApplication</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showApplicationAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOptions</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showPreferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpenProject</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewRule</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>newRule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewProgram</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>newProgram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNewGraph</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>newGraph()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveCurrentFile</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAll</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveCurrentFileAs</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLayoutCircular</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>layoutCircular()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>43</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLayoutSugiyama</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>layoutSugiyama()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showApplicationHelp()</slot>
  <slot>newProject()</slot>
  <slot>showPreferences()</slot>
  <slot>showApplicationAbout()</slot>
  <slot>openProject()</slot>
  <slot>openProject(QString)</slot>
  <slot>newRule()</slot>
  <slot>newProgram()</slot>
  <slot>newGraph()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
  <slot>saveAll()</slot>
  <slot>layoutSugiyama()</slot>
  <slot>layoutCircular()</slot>
 </slots>
</ui>
